walrus_mainboard_driver/src/walrus_mainboard_driver/walrus_mainboard_driver.cpp

```diff
@@ -0,0 +1,670 @@
+		        if (temp < 0)
+		            temp += 2*PI;
+		        dr = temp - pod_position[l];
+		        if (dr > 0 && dr > PI) //Decrease angle across 0
+		            pod_velocity[l] = (dr-2*PI)/interval;
+		        else if (dr < 0 && dr < -PI) //Increase angle across 0
+		            pod_velocity[l] = (dr+2*PI)/interval;
+		        else  
+		            pod_velocity[l] = dr/interval;
+		        pod_position[l] = temp;		
+		        pod_effort[l] = (hs_feedback_msg.motor_current[l] / 1000.0) * 29.184; //RS550 motors with 5120:1 reduction outputs 29.185 Nm/A
+	        }		
+	    }
+
+	}
+	
+	void MainBoardDriver::write()
+	{			
+	    walrus_firmware_msgs::MainBoardHighSpeedControl hs_control_msg;    
+	    {
+	        boost::lock_guard<boost::mutex> lock(control_data_mutex);
+	        //Send motor control message	        
+	        for (int l = 0; l < 4; l++)
+	        {
+	            pod_power[l] = (pod_effort_cmd[l]*250) + 1500;
+		        hs_control_msg.motor_power[l] = pod_power[l];
+		    }
+		}
+		
+		hs_control.publish(hs_control_msg);		
+	}
+	
+	void MainBoardDriver::update_diagnostics()
+	{
+		diagnostic_updater.update();
+	}
+	
+	//Subscription callbacks
+	void MainBoardDriver::hs_feedback_callback(const walrus_firmware_msgs::MainBoardHighSpeedFeedback& msg)
+	{
+	    boost::lock_guard<boost::mutex> lock(control_data_mutex);
+	    hs_feedback_msg = msg;
+	    last_hs_msg = ros::Time::now();
+	}
+	void MainBoardDriver::ls_data_callback(const walrus_firmware_msgs::MainBoardLowSpeedData& msg)
+	{
+	    boost::lock_guard<boost::mutex> lock(sensor_data_mutex);
+        
+        //Temp sensors
+        for (int l = 0; l < 10; l++)
+            temp_sensors[l] = msg.temp_sense[l] / 10.0;
+        
+        //Ambient temp
+        ambient_temp = msg.board_temp / 100.0;
+        
+        //Humidity
+        humidity = msg.humidity / 100.0;
+        
+        //Pressure
+        pressure = msg.pressure / 1000.0;
+        
+        //Water sensors
+        water[0] = msg.water_sense & 1;
+        water[1] = msg.water_sense & 2;
+        water[2] = msg.water_sense & 4;
+        water[3] = msg.water_sense & 8;
+        water[4] = msg.water_sense & 16;
+        water[5] = msg.water_sense & 32;
+        
+        //Backup battery
+        backup_voltage = msg.tension[BACKUP_BATT_VOLTAGE];
+        
+        int count = 0;
+        main_voltage = 0;
+        main_current = 0;
+        main_avg_current = 0;
+        main_charge = 0;
+        //Battery data
+        for (int l = 0; l < 4; l++)
+        {
+            if (msg.lcell_charge[l] > 0)
+            {
+                //if it wasn't present, ask it for mfr and name data
+                if (!batteries[l].present)
+                {
+                    batteries[l].mfr = "Not available";
+                    batteries[l].dev_name = "Not available";
+                    batteries[l].chemistry = "Not available";
+                    walrus_firmware_msgs::MainBoardControl msg2;
```

Comment:
since you have multiple messages can you name them something useful instead of msg and msg2
like sensor_msg and control_msg or something along those lines